source/turtles/turtle_hero.py

```diff
@@ -0,0 +1,137 @@
+import pygame, random
+
+GREEN = (20, 255, 140)
+GREY = (210, 210, 210)
+WHITE = (255, 255, 255)
+RED = (255, 0, 0)
+PURPLE = (255, 0, 255)
+BLACK = (0, 0, 0)
+WIDTH = 100
+HEIGHT = 50
+ACC = 600
+TICK = 1/60.0
+SPEED_FAST = 300
+SPPED_SLOW = 150
+
+class Turtle(pygame.sprite.Sprite):
+    # This class represents a turtle. It derives from the "Sprite" class in Pygame.
+
+    def __init__(self, type, size_coeff, name, position):
+        """
+        :param type: turtle type (TODO: define)
+        :param size_coeff: SIZE_COEFF
+        :param name: name strinf
+        :param position: creation postition in pixels - tuple
+        """
+        # Call the parent class (Sprite) constructor
+        super().__init__()
+
+        # Pass in the name of the turtle, type, size, and its x and y position
+        # Set the background color and set it to be transparent
+        self.name = name
+        self.image = pygame.Surface([WIDTH * size_coeff, HEIGHT * size_coeff])
+        self.image.fill(WHITE)
+        self.image.set_colorkey(WHITE)
+
+        self.size = size_coeff
+        self.type = type
+        # Draw the turtle (a rectangle!)
+        pygame.draw.rect(self.image, BLACK, [0, 0, WIDTH * size_coeff, HEIGHT * size_coeff])
+        # Instead we could load a proper picture of a turtle...
+        # self.image = pygame.image.load("turtle.png").convert_alpha()
+
+        # Fetch the rectangle object that has the dimensions of the image and set position.
+        self.rect = self.image.get_rect()
+        self.rect.x = float(position[0])
+        self.tmp_x_float = float(position[0])
+        self.rect.y = position[1]
+
+    def move_right(self, pixels):
+        self.rect.x += pixels
+
+    def move_left(self, pixels):
+        self.rect.x -= pixels
+
+    def move_up(self, pixels):
+        self.rect.y -= pixels
+
+    def move_down(self, pixels):
+        self.rect.y += pixels
+
+
+class TurtleHero(Turtle):
+    def __init__(self, type, size_coeff, name, position):
+        super().__init__(type, size_coeff, name, position)
+        self.is_jumping = 0
+        self.dist_to_jump = 0
+        self.initial_y = 0
+        self.jump_counter = 0
+        self.moving = 0
+        self.speed = 0
+        self.speed_act = 0
+        self.speed_target = 0
+        self.tmp_x_float = 0
+
+    def init_jump(self, initial_v, grav_acc):
+        self.is_jumping = 1
+        self.dist_to_jump = (initial_v) ** 2 / (2 * grav_acc)
+        print("init jump")
+        print(self.dist_to_jump)
+        self.initial_y = self.rect.y
+        self.jump_counter = self.jump_counter + 1
+
+    def changeSpeed(self, speed):
+        self.speed = speed
+
+    def init_move_right(self):
+        self.speed_target = SPPED_SLOW
```

Comment:
nawiązanie do pierwszego komentarza, pamiętaj o tym, zeby nie korzystac z takich globali bez potrzeby
jeśli to konieczne, to najlepiej wydziel je gdzies do osobnego pliku czy tak jak tutaj masz (na górze), ale i tak poprzypisuj w konstruktorze jako atrybuty klasy